code/datums/diseases/_MobProcs.dm

```diff
@@ -32,23 +29,30 @@
 
 
 /mob/proc/AddDisease(datum/disease/D)
-	var/datum/disease/DD = new D.type(1, D, 0)
-	viruses += DD
-	DD.affected_mob = src
-	DD.holder = src
-
-	//Copy properties over. This is so edited diseases persist.
-	var/list/skipped = list("affected_mob","holder","carrier","stage","type","parent_type","vars","transformed")
-	for(var/V in DD.vars)
-		if(V in skipped)
-			continue
-		if(istype(DD.vars[V],/list))
-			var/list/L = D.vars[V]
-			DD.vars[V] = L.Copy()
-		else
-			DD.vars[V] = D.vars[V]
-
-	DD.affected_mob.med_hud_set_status()
+	for(var/datum/disease/advance/P in viruses)
+		if(istype(D, /datum/disease/advance))
+			var/datum/disease/advance/DD = D
+			if (P.totalResistance() < DD.totalTransmittable()) //Overwrite virus if the attacker's Transmission is lower than the defender's Resistance. This does not grant immunity to the lost virus.
+				P.remove_virus()
```

Comment:
if we can only ever have one virus at a time, then the viruses var no longer needs to be a list.